pandas/core/resample.py

```diff
@@ -1693,11 +1693,15 @@ def _get_timestamp_range_edges(
     -------
     A tuple of length 2, containing the adjusted pd.Timestamp objects.
     """
-    index_tz = first.tz
-    if isinstance(origin, Timestamp) and (origin.tz is None) != (index_tz is None):
-        raise ValueError("The origin must have the same timezone as the index.")
-
     if isinstance(freq, Tick):
+        index_tz = first.tz
```

Comment:
shouldn't this actually be close to where you set the origin (e.g. in the TimeGrouper)?

Reply:
I don't have access to the variables `first` or `end` to infer the index timezone in `TimeGrouper.__init__` so I don't think I can do what you suggest. :confused: 

We used to infer the `index_tz` before #31809 as you can see here on the tag `v.1.0.4`:
https://github.com/pandas-dev/pandas/blob/1ce1c3c1ef9894bf1ba79805f37514291f52a9da/pandas/core/resample.py#L1650

In my opinion that make sense since the API of `groupby` (shared with the `resample` API) can accept any kind of grouper, so the bins (that needs the information of `first` and `end`) can only be calculated when we apply that against a Series or a DataFrame.

```python
import pandas as pd
import numpy as np

start, end = "2000-08-02 23:30:00+0500", "2000-12-02 00:30:00+0500"
rng = pd.date_range(start, end, freq="7min")
ts_1 = pd.Series(np.random.randn(len(rng)), index=rng)
ts_2 = pd.Series(np.random.randn(len(rng)), index=rng)

# => The example line that create this impossibility:
grouper = pd.Grouper(freq="D", origin="epoch") 
result_1 = ts_1.groupby(grouper).mean()
result_2 = ts_2.groupby(grouper).mean()
```